src/app/posts/components/post-form/post-form.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { PostFormService } from '../../services/post-form.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-post-form',
+  templateUrl: './post-form.component.html',
+  styleUrls: ['./post-form.component.scss']
+})
+
+export class PostFormComponent{
+  postForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder, private postService: PostFormService, private modal: NgbModal) {
+      this.postForm = this.formBuilder.group({
+        author: ['', [Validators.required, Validators.minLength(4), Validators.maxLength(50)]],
+        title: ['', [Validators.required, Validators.minLength(5)]],
+        content: ['', [Validators.required, Validators.minLength(10)]]
+      })
+   }
+
+  onSubmit() {
+    if(this.postForm.status === 'VALID') {
```

Comment:
there are should be a  special filed valid in postForm

Reply:
Fixed in new PR Feature/hw3, thanks